### PR TITLE
add contracts with quadratic voting

### PR DESCRIPTION
**Pull request by byteZorvin**

### CODE REVIEW

File: contracts/contracts/VoteModule.sol [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+
+        // require(isDomainVoted[domain][msg.sender] == false, "Domain already voted");
+
+        require(isLegit != domainVerdict[domain].isScam, "Conflicting report");
+
+        string memory evidenceHashesSerialised = "";
+        uint256 nEvidences = evidenceHashes.length;
+        require(nEvidences > 0, "Require atleast one evidence");
+
+        for (uint256 i = 0; i < nEvidences; ++i) {
```
Comment (VenkatTeja) [inline contracts/contracts/VoteModule.sol:237]: Can ignore comments and evidences for reporting. need only if requesting validation.

File: contracts/contracts/VoteModule.sol [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+        if (isApproved) {
+            claim.status = "Approved";
+            claim.domainVerdictScamOrNot = domainVerdictScamOrNot;
+            claim.validatorComments = _validatorComments;
+            rewardTokenContract.mint(claim.requestor, rewardAmount);
+        } else {
+            claim.status = "Rejected";
+        }
+
+        uint256 totalVotesToSlash = validationFees / votePrice;
```
Comment (VenkatTeja) [inline contracts/contracts/VoteModule.sol]: be careful with math here

File: contracts/contracts/VoteModule.sol (hunk 1 above)
Comment (byteZorvin) [reply contracts/contracts/VoteModule.sol:237]: Fixed

File: contracts/contracts/VoteModule.sol (hunk 2 above)
Comment (byteZorvin) [reply contracts/contracts/VoteModule.sol]: Checked again, fixed

Comment (byteZorvin) [reply contracts/contracts/VoteModule.sol]: Checked again, fixed

**Pull request merged**